extensions/interactions/MusicNotesInput/MusicNotesInput.js

```diff
@@ -150,15 +150,49 @@ oppia.directive('oppiaInteractiveMusicNotesInput', [
         var noteChoicesElt = $element.find('.oppia-music-input-note-choices');
         var staffContainerElt = $element.find('.oppia-music-input-staff');
 
-        // Sets grid positions and initializes the view after staff has loaded.
-        setTimeout(function() {
-          $scope.init();
-        }, 1000);
+        // Sets grid positions, displays the staff and note,
+        // and then initializes the view after staff has loaded.
+        $(document).ready(function() {
+          setTimeout(function() {
+            // Display valid note area only after page has fully loaded.
+            $('.oppia-music-input-valid-note-area').css('visibility', 'visible');
+            $scope.init();
+          }, 20);
+        });
+
+        // Staff has to be reinitialized every time that the staff is resized or
+        // displayed. The staffContainerElt and all subsequent measurements
+        // must be recalculated in order for the grid to work properly.
+        $scope.reinitStaff = function() {
+          // Prevent staff from reinitializing unneccesarily.
+          if (!$('.img-circle').data('fired')) {
+            $('.oppia-music-input-valid-note-area').css('visibility', 'hidden');
+            var that = $('.img-circle');
+            setTimeout(function() {
+              $(that).data('fired', true);
+              $('.oppia-music-input-valid-note-area').css('visibility', 'visible');
+              $scope.init();
+            }, 200);
+          }
+        };
+
+        // TODO(wagnerdmike): This would be better if it listened for a
+        // card switch event instead of the button click. Reinit needs to be
+        // triggered when page is in the smaller one card format or else
+        // dimensions for the staff will be incorrectly calculated.
+        $('.img-circle').on("click", function(e) {
```

Comment:
Thanks for the tip here. I added a listener to showInteraction and now I don't need to reinit on resize anymore because the parent div doesn't change anymore like it did when it was first designed. Works much better I think.